src/typings/goodchat.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Message Content Object - Common properties
+ *
+ * @export
+ * @interface MessageContentBase
+ */
+ export interface MessageContentBase {
+  type: string
+}
+
+/**
+ * Message Text Record
+ *
+ * @export
+ * @interface MessageTextContent
+ * @extends {MessageContentBase}
+ */
+export interface MessageTextContent extends MessageContentBase {
+  type: "text"
+  text: string
+}
+
+/**
+ * Message Image record
+ *
+ * @export
+ * @interface MessageImageContent
+ * @extends {MessageContentBase}
+ */
+export interface MessageImageContent extends MessageContentBase {
+  type:       "image"
+  mediaUrl:   string
+  mediaType:  string,
+  mediaSize:  number,
+  altText?:   string
+}
+
+export type MessageContent = MessageTextContent | MessageImageContent
+
+export type MessageContentType = "image" | "text"
+
+export enum AuthorType {
+  STAFF = 'STAFF',
+  CUSTOMER = 'CUSTOMER',
+  SYSTEM = 'SYSTEM'
+};
+
+export * from '../generated/graphql'
```

Comment:
Why re-export generated types?
Also, any problem with consumers importing from the generated file directly instead of from `typings`?

Reply:
not a "problem", but it serves a purpose to re-export them as we now have control over each type. So we're not dependant on the generated types